### PR TITLE
Fixes #77

### LINKED ISSUE #77: Mvc.CustomRoutingConvention is broken
Upon startup:

```
System.TypeLoadException: Method 'get_Configuration' in type 'Microsoft.AspNet.H
osting.HostingEnvironment' from assembly 'Microsoft.AspNet.Hosting, Version=1.0.
0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' does not have an implemen
tation.
   at Microsoft.AspNet.Hosting.WebHostBuilder..ctor(IConfiguration config, Boole
an captureStartupErrors)
   at Microsoft.AspNet.Hosting.WebApplication.Run(Type startupType, String[] arg
s)
   at Microsoft.AspNet.Hosting.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly,
 String[] args, IServiceProvider serviceProvider)
   at Microsoft.Dnx.ApplicationHost.Program.<>c__DisplayClass3_0.<ExecuteMain>b_
_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
```

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes Mvc.CustomRoutingConvention and adds a test to make sure it keeps working. Fixes #77

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
@ryanbrandenburg, you'll need to rebase on my changes since we have changed our Hosting API. We are moving towards using only Kestrel. Take a look at the other examples and let me know if you have any questions!

**Comment by ryanbrandenburg** [comment]
cc @rynowak I was told you were a good candidate to review this.

File: samples/Mvc.CustomRoutingConvention/project.json [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
     "emitEntryPoint": true
   },
-  "commands": {
-    "web": "Mvc.CustomRoutingConvention"
+  "dependencies": {
+    "Microsoft.AspNet.Hosting": "1.0.0-*",
+    "Microsoft.AspNet.Mvc": "6.0.0-*",
+    "Microsoft.AspNet.Server.IIS": "1.0.0-*",
+    "Microsoft.AspNet.Server.Kestrel": "1.0.0-*",
+    "Microsoft.AspNet.Server.WebListener": "1.0.0-*"
```
Comment (rynowak) [inline samples/Mvc.CustomRoutingConvention/project.json]: Why do we need all of this stuff?

File: test/FunctionalTests/MvcTests/CustomRoutingConventionTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6400")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.Mono, RuntimeArchitecture.x86, "http://localhost:6500")]
+        public async Task RunSite_Other(
+            ServerType server,
+            RuntimeFlavor runtimeFlavor,
+            RuntimeArchitecture architecture,
+            string applicationBaseUrl)
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
```
Comment (rynowak) [inline test/FunctionalTests/MvcTests/CustomRoutingConventionTests.cs]: Why are any of these variations relevant? It should be possible to just run this as an in memory test.

Comment (ryanbrandenburg) [inline test/FunctionalTests/MvcTests/CustomRoutingConventionTests.cs]: I'm just aping the other functional tests in this repo. Do you know where I can find an example of testing routes in memory?

Comment (rynowak) [inline test/FunctionalTests/MvcTests/CustomRoutingConventionTests.cs]: You should just be able to use TestServer if you replace the assembly provider to special case this assembly

**Comment by muratg** [comment]
This looks good.

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**